matlab/src/cpp/arrow/matlab/array/proxy/string_array.cc

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/matlab/array/proxy/string_array.h"
+
+#include "arrow/array/builder_binary.h"
+
+#include "arrow/matlab/error/error.h"
+#include "arrow/matlab/bit/pack.h"
+#include "arrow/matlab/bit/unpack.h"
+#include "arrow/util/utf8.h"
+
+namespace arrow::matlab::array::proxy {
+
+        libmexclass::proxy::MakeResult StringArray::make(const libmexclass::proxy::FunctionArguments& constructor_arguments) {
+            namespace mda = ::matlab::data;
+
+            mda::StructArray opts = constructor_arguments[0];
+            const mda::StringArray array_mda = opts[0]["MatlabArray"];
+            const mda::TypedArray<bool> unpacked_validity_bitmap_mda = opts[0]["Valid"];
+
+            // Convert UTF-16 encoded MATLAB string values to UTF-8 encoded Arrow string values.
+            const auto array_length = array_mda.getNumberOfElements();
+            std::vector<std::string> strings;
+            strings.reserve(array_length);
+            for (const auto& str : array_mda) {
+                // Substitute MATLAB string(missing) values with the empty string value ("") before converting to Arrow format.
+                const auto str_utf16 = str ? *str : std::u16string{u""};
+                MATLAB_ASSIGN_OR_ERROR(auto str_utf8, arrow::util::UTF16StringToUTF8(str_utf16), error::UNICODE_CONVERSION_ERROR_ID);
+                strings.push_back(str_utf8);
```

Comment:
Can we use `std::move()` here to avoid copying?

```suggestion
                strings.push_back(std::move(str_utf8));
```